### DISCUSSION TITLE
Etcd internal rpc call failed（grpc_server_handled_total{grpc_code="OutOfRange",grpc_method="Range",grpc_service="etcdserverpb.KV",grpc_type="unary"}）

### DISCUSSION THREAD

**Discussion by my1990** [etcd-io/etcd#15108] [General]
I received an alarm that the internal rpc call of etcd failed. It is about etcdserverpb.KV
![image](https://user-images.githubusercontent.com/16768690/212584520-adf788fc-e20b-4dde-8cb8-2b5a88949e14.png)

grpc_server_handled_total{grpc_service="etcdserverpb.KV",grpc_code="OutOfRange",grpc_method="Range"}  
![image](https://user-images.githubusercontent.com/16768690/212585281-46ebbd84-e903-4885-a7fa-7ae059171d2d.png)

I know that Etcd is stored as KV and will retain the historical version for each key. If we specify a non-existent kv revision, an error will be reported. At this time, etcd will record an OutOfRange. But so many OutOfRange records are certainly not triggered manually. All those who want to know the mechanism of etcd will generate these OutOfRange records

**Comment by ahrtr** [comment]
>  If we specify a non-existent kv revision, an error will be reported. 

It's expected behavior. etcd will report `ErrGRPCFutureRev` if the rev is bigger than current rev, and `ErrGRPCCompacted` if it's been compacted. Why do you specify a non-exist rev?

**Reply by my1990** [reply]
In fact, we did not specify rev, but there are still many OutOfRange errors in etcd, which are increasing slowly. All want to know where the error comes from.

I checked the log and found that the following errors are related to disk performance?

![image](https://user-images.githubusercontent.com/16768690/212591602-56a24d91-22fd-452a-bdc4-b3b2b9dc5946.png)

![image](https://user-images.githubusercontent.com/16768690/212591603-45f8f1a6-45ab-43e9-a85e-d907b0b2dc73.png)

**Comment by my1990** [comment]
After testing, manually trigger OutOfRange, and there will be no records of ErrGRPCFutureRev and ErrGRPCCompacted in the etcd log. So the OutOfRange error should be caused by some internal mechanism of the system

<img width="943" alt="09387466-A485-4e8c-8B8C-7B7F0230E382" src="https://user-images.githubusercontent.com/16768690/212593851-2b152e4a-1ca3-4b7f-95b2-3271af57988a.png">

**Comment by ahrtr** [comment]
Please follow [sanity_check_and_collect_basic_info.md](https://github.com/ahrtr/etcd-issues/blob/master/docs/troubleshooting/sanity_check_and_collect_basic_info.md) to provide the requested info.
